ScanServerIT.testScanWithTabletHostingMix is failing

The test is scanning rows `-inf` to `row_0000000003`. The test is failing with:

```
java.lang.IllegalStateException: org.apache.accumulo.core.client.AccumuloException: org.apache.accumulo.core.client.InvalidTabletHostingRequestException: Extent 1;row_0000000004;row_0000000003 has a tablet hosting goal state NEVER
	at org.apache.accumulo.core.clientImpl.ScannerIterator.getNextBatch(ScannerIterator.java:187)
	at org.apache.accumulo.core.clientImpl.ScannerIterator.hasNext(ScannerIterator.java:111)
	at com.google.common.collect.Iterators.size(Iterators.java:168)
	at com.google.common.collect.Iterables.size(Iterables.java:129)
	at org.apache.accumulo.test.ScanServerIT.testScanWithTabletHostingMix(ScanServerIT.java:250)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.junit.platform.commons.util.ReflectionUtils.invokeMethod(ReflectionUtils.java:727)
	at org.junit.jupiter.engine.execution.MethodInvocation.proceed(MethodInvocation.java:60)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain$ValidatingInvocation.proceed(InvocationInterceptorChain.java:131)
	at org.junit.jupiter.api.AssertTimeoutPreemptively.lambda$submitTask$3(AssertTimeoutPreemptively.java:95)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:829)
Caused by: org.apache.accumulo.core.client.AccumuloException: org.apache.accumulo.core.client.InvalidTabletHostingRequestException: Extent 1;row_0000000004;row_0000000003 has a tablet hosting goal state NEVER
	at org.apache.accumulo.core.clientImpl.ThriftScanner.scan(ThriftScanner.java:816)
	at org.apache.accumulo.core.clientImpl.ScannerIterator.readBatch(ScannerIterator.java:159)
	at org.apache.accumulo.core.clientImpl.ScannerIterator.getNextBatch(ScannerIterator.java:177)
	... 16 more
Caused by: org.apache.accumulo.core.client.InvalidTabletHostingRequestException: Extent 1;row_0000000004;row_0000000003 has a tablet hosting goal state NEVER
	at org.apache.accumulo.core.clientImpl.ClientTabletCacheImpl.requestTabletHosting(ClientTabletCacheImpl.java:680)
	at org.apache.accumulo.core.clientImpl.ClientTabletCacheImpl.findTablet(ClientTabletCacheImpl.java:569)
	at org.apache.accumulo.core.clientImpl.ThriftScanner.getNextScanAddress(ThriftScanner.java:559)
	at org.apache.accumulo.core.clientImpl.ThriftScanner.scan(ThriftScanner.java:662)
	... 18 more
```

I'm thinking that the Scanner is trying to host the next tablet preemptively.

--- a/core/src/main/java/org/apache/accumulo/core/clientImpl/ClientTabletCacheImpl.java
+++ b/core/src/main/java/org/apache/accumulo/core/clientImpl/ClientTabletCacheImpl.java
@@ -600,7 +600,8 @@ public class ClientTabletCacheImpl extends ClientTabletCache {
       var currTablet = extent;
 
       for (int i = 0; i < hostAheadCount; i++) {
-        if (currTablet.endRow() == null || !hostAheadRange.contains(new Key(currTablet.endRow()))) {
+        if (currTablet.endRow() == null || hostAheadRange
+            .afterEndKey(new Key(currTablet.endRow()).followingKey(PartialKey.ROW))) {
           break;
         }
 

